src/s3/commands/uploadFile.ts

```diff
@@ -26,53 +38,126 @@ export interface FileSizeBytes {
 }
 
 /**
- * Uploads a file to the bucket or folder represented by the given node.
- *
- * Prompts the user for the file location.
- * Shows the output channel with "upload started" message.
- * Uploads the file (showing a progress bar).
- * Shows the output channel with "upload completed" message.
- * Refreshes the node.
- *
- * Node that the node is reset to displaying its first page of results.
- * The file that is uploaded won't necessary fall on the first page.
- * The user may need to load more pages to see the uploaded file reflected in the tree.
+ * Wizard to upload a file.
+ * Either a node or the document has to be given, will follow different path depending on what is given
+ * 
+ * @param s3Client account to upload the file to
+ * @param nodeOrDocument node to upload to or file currently open
+ * 
  */
-export async function uploadFileCommand(
-    node: S3BucketNode | S3FolderNode,
+ export async function uploadFileCommand(
+    s3Client: S3Client,
+    nodeOrDocument: S3BucketNode | S3FolderNode | vscode.Uri | undefined,
     fileSizeBytes: FileSizeBytes = statFile,
+    getBucket = promptUserForBucket,
+    getFile = getFileToUpload,
     window = Window.vscode(),
-    commands = Commands.vscode(),
-    outputChannel = ext.outputChannel
-): Promise<void> {
-    getLogger().debug('UploadFile called for %O', node)
-
-    const fileLocation = await promptForFileLocation(window)
-    if (!fileLocation) {
-        getLogger().info('UploadFile cancelled')
-        telemetry.recordS3UploadObject({ result: 'Cancelled' })
-        return
+    outputChannel = ext.outputChannel,
+    commands = Commands.vscode()
+): Promise<void>{
+    let key: string
+    let bucket: S3.Bucket
+    let file: vscode.Uri | undefined
+    let node: S3BucketNode | S3FolderNode | undefined
+    let document: vscode.Uri | undefined
+
+    if (nodeOrDocument){
+        const isNode = !!((nodeOrDocument as any).getChildren)
```

Comment:
Nit: unnecessary const, make this assignment the condition for the if statement below.